dist/init-scripts/upstart.erb

```diff
@@ -0,0 +1,24 @@
+# Upstart job for <%= @service_name %>
+
+description "<%= @description ? @service_name + ' - ' + @description : @service_name %>"
+
+start on (local-filesystems and net-device-up IFACE!=lo)
+stop on runlevel [!2345]
+
+# up ulimit -n a bit
+limit nofile 10000 10000
+
+setuid "<%= @service_name %>"
+setgid "<%= @service_name %>"
+
+env NODE_PATH="/srv/deployment/<%= @service_name %>/deploy/node_modules"
+env <%= @service_name.gsub(/[^a-zA-Z0-9_]/, '_').upcase %>_PORT="<%= @port %>"
+
+respawn
+
+# wait 60 seconds for a graceful restart before killing the master
+kill timeout 60
+
+chdir /srv/deployment/<%= @service_name %>/deploy
+exec /usr/bin/nodejs src/server.js -c /etc/<%= @service_name %>/config.yaml >> /var/log/<%= @service_name %>/main.log 2>&1
```

Comment:
> IMHO we should also drop the upstart config completely, as the init script will work on Ubuntu as well, and Upstart is on its way out. Maintaining only one of the two avoids duplicate work.

I'd be in favour of that option too. Upstart can use SysV init scripts, and that plays well with our [base::service_unit](https://github.com/wikimedia/operations-puppet/blob/production/modules/base/manifests/service_unit.pp) module in puppet too. Another point in favour of the drop is that getting upstart to log to syslog is [rather tricky](http://serverfault.com/a/666970) (a simple pipe won't do it, as then Upstart picks up the PID of `logger`).

@filippog thoughts?